client/pages/detail.jsx

```diff
@@ -0,0 +1,194 @@
+import React from 'react';
+import Navbar from '../components/navbar';
+import Loading from './loading';
+
+class detail extends React.Component {
+  constructor(props) {
+    const device = window.innerWidth < 768 ? 'small' : 'regular';
+    super(props);
+    this.state = {
+      intervalId: null,
+      carouselIndex: 0,
+      game: null,
+      screenshots: null,
+      device
+    };
+  }
+
+  componentDidMount() {
+    const gameId = this.props.id;
+
+    fetch(`http://localhost:3000/api/game/${gameId}`)
+      .then(r => r.json())
+      .then(r => {
+        this.setState({ game: r });
+      })
+      .catch(r => console.error(r));
+
+    fetch(`http://localhost:3000/api/screenshots/${gameId}`)
+      .then(r => r.json())
+      .then(r => {
+        this.setState({ screenshots: r }, this.autoscroll);
+      })
+      .catch(r => console.error(r));
+
+    window.addEventListener('resize', e => {
+      const device = window.innerWidth < 768 ? 'small' : 'regular';
+      if (this.state.device !== device) {
+        this.setState({ device });
+      }
+    });
+  }
+
+  autoscroll() {
+    const { intervalId } = this.state;
+    clearInterval(intervalId);
+
+    const newIntervalId = setInterval(() => {
+      const currentIndex = this.convertIndex(this.state.carouselIndex, 'next');
+      this.setState({ carouselIndex: currentIndex });
+    }, 3000);
+    this.setState({ intervalId: newIntervalId });
+  }
```

Comment:
If a page component adds a window event listener or starts an interval, it's important to remove or cancel those when the view swaps to another page, so we don't leak memory

```js
window.addEventListener('resize', this.handleResize);

componentWillUnmount() {
  window.removeEventListener('resize', this.handleResize);
  clearInterval(this.interval);
}
```

https://reactjs.org/docs/react-component.html#componentwillunmount